spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/health/DatastoreHealthIndicator.java

```diff
@@ -32,28 +34,29 @@
  * @author Raghavan N S
  * @author Srinivasa Meenavalli
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  *
  * @since 1.2
  */
 @Component
 public class DatastoreHealthIndicator extends AbstractHealthIndicator {
 
-	private final Datastore datastore;
+	private final Supplier<Datastore> datastore;
```

Comment:
This is a bit unfortunate because we will probably break other users' code in a similar way.

Reply:
If they were expecting to get a bean of Datastore directly, then yes we will break them :(